watch/watch.go

```diff
@@ -62,12 +75,33 @@ func ParseExempt(params map[string]interface{}, exempt []string) (*Plan, error)
 	if err := assignValue(params, "type", &plan.Type); err != nil {
 		return nil, err
 	}
-
 	// Ensure there is a watch type
 	if plan.Type == "" {
 		return nil, fmt.Errorf("Watch type must be specified")
 	}
 
+	// Get the specific handler
+	if err := assignValue(params, "handler_type", &plan.HandlerType); err != nil {
+		return nil, err
+	}
+	switch plan.HandlerType {
+	case "http":
+		if _, ok := params["http_handler_config"]; !ok {
+			return nil, fmt.Errorf("Handler type 'http' requires 'http_handler_config' to be set")
+		}
+		config, err := parseHttpHandlerConfig(params["http_handler_config"])
+		if err != nil {
+			return nil, fmt.Errorf(fmt.Sprintf("Failed to parse 'http_handler_config': %v", err))
+		}
+		plan.Exempt["http_handler_config"] = config
+		delete(params, "http_handler_config")
+
+	case "":
```

Comment:
Once you convert empty to "script" before this is ever called, you can turn this into "script"